### PR TITLE
tutanota-error package, remove keytar, update electron

### LINKED ISSUE #6342: Reconcile tutanota-crypto CryptoError and TutanotaError CryptoError
There's been a todo there for a while and it's become a source of errors.

I think the easiest way to do this might be to extract an error package (or extract the errors and their support cast into tutanota-utils) and depend on it from tutanota-crypto.

We also might move to comparing names instead of classes in general for errors since we have a similar problem with the IPC

### PR DESCRIPTION
**Pull request by charlag**
Introduces a new package in order to avoid tutanota-crypto depending on the whole tutanota-utils. With this package we can have one canonical CryptoError.

close #6342

### CODE REVIEW

**Comment by charlag** [comment]
Something about clean install doesn't work yet, I'll look into it

**Comment by charlag** [comment]
Sorry for the big initial commit, I had to come back to it and fix things multiple times before it worked.

On the upside we now have more proper imports

File: packages/tutanota-error/lib/TutanotaError.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+	constructor(name: string, message: string) {
+		super(message)
+		this.name = name
+		this.message = message
+
+		if (typeof Error.captureStackTrace === "function") {
+			Error.captureStackTrace(this, this.constructor)
+		} else {
+			let error = new Error()
```
Comment (ganthern) [inline packages/tutanota-error/lib/TutanotaError.ts:36]: can we assume that stacks are always set by now? at least IE9 should be irrelevant today right?

File: src/api/main/EventController.ts [hunk 2]
```diff
@@ -1,39 +1,19 @@
...
-		isUpdateForTypeRef(typeRef, update) &&
-		(update.instanceListId === "" ? isSameId(update.instanceId, entity._id) : isSameId([update.instanceListId, update.instanceId], entity._id))
-	)
-}
 
 export type ExposedEventController = Pick<EventController, "onEntityUpdateReceived" | "onCountersUpdateReceived">
 
 const TAG = "[EventController]"
 
+export type EntityEventsListener = (updates: ReadonlyArray<EntityUpdateData>, eventOwnerGroupId: Id) => Promise<any>
```
Comment (ganthern) [inline src/api/main/EventController.ts:15]: `any` -> `unknown`?

**Review by ganthern — COMMENTED** [review]
I feel like we should mention _somewhere_ that all errors that extend `TutanotaError`
should only be exported from a dedicated entry point to make it impossible to accidentally import a whole package when only the error is asked for.

File: packages/tutanota-error/lib/TutanotaError.ts (hunk 1 above)
Comment (charlag) [reply packages/tutanota-error/lib/TutanotaError.ts]: I suspect it's more than IE9 and I didn't want to take this risk

File: src/api/common/utils/ErrorUtils.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// @bundleInto:common
```
Comment (charlag) [inline src/api/common/utils/ErrorUtils.ts:1]: isn't it already in api/common?

**Pull request merged**